File: java/Problem016.java
import java.math.BigInteger;

public class Problem016 {
    public static void main(String[] args) {
        // 2^1000, each character mapped to integer value, and summed up.
        System.out.println(BigInteger.valueOf(2).pow(1000).toString().chars().mapToLong(Character::getNumericValue).sum());
    }
}

File: kotlin/Problem016.kt
import java.math.BigInteger
import java.util.stream.IntStream

fun main(args: Array<String>) {
    EulerSolution.execute(Problem016())
}

// cool class extension
fun Number.digits() : IntStream {
    return this.toString().chars().map(Character::getNumericValue)
}

class Problem016 : EulerSolution {
    override fun run() : Long {
        return BigInteger.valueOf(2).pow(1000).digits().sum().toLong()
    }
}
